### PR TITLE
Sleep before genesis

### PR DESCRIPTION
**Pull request by nholland94**

### CODE REVIEW

File: src/lib/proposer/proposer.ml [hunk 1]
```diff
@@ -484,15 +484,33 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
...
+         * erase this timeout even if the last run of the proposer wants to wait
+         * for a long while.
+         * *)
+        Agent.on_update keypairs ~f:(fun _new_keypairs ->
+            Singleton_scheduler.schedule scheduler (Time.now time_controller)
+              ~f:check_for_proposal ) ;
+        check_for_proposal ()
+      in
+      let now = Time.now time_controller in
+      if Time.( < ) now Consensus.Constants.genesis_state_timestamp then
```
Comment (bkase) [inline src/lib/proposer/proposer.ml]: flip the bool

File: src/lib/proposer/proposer.ml [hunk 2]
```diff
@@ -484,15 +484,33 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
...
+         * Given that rescheduling takes the min of the two timeouts, we won't
+         * erase this timeout even if the last run of the proposer wants to wait
+         * for a long while.
+         * *)
+        Agent.on_update keypairs ~f:(fun _new_keypairs ->
+            Singleton_scheduler.schedule scheduler (Time.now time_controller)
+              ~f:check_for_proposal ) ;
+        check_for_proposal ()
+      in
+      let now = Time.now time_controller in
```
Comment (bkase) [inline src/lib/proposer/proposer.ml]: can you stick a comment here explaining why there's this logic here before starting?

File: src/lib/proposer/proposer.ml [hunk 3]
```diff
@@ -484,15 +484,33 @@ let run ~logger ~prover ~verifier ~trust_system ~get_completed_work
...
-       *
-       * Given that rescheduling takes the min of the two timeouts, we won't
-       * erase this timeout even if the last run of the proposer wants to wait
-       * for a long while.
-       * *)
-      Agent.on_update keypairs ~f:(fun _new_keypairs ->
-          Singleton_scheduler.schedule scheduler (Time.now time_controller)
-            ~f:check_for_proposal ) ;
-      check_for_proposal () )
+      let start _ =
```
Comment (bkase) [inline src/lib/proposer/proposer.ml]: why not `start ()`

Comment (nholland94) [reply src/lib/proposer/proposer.ml]: I just wanted to be able to pass it to `Time.Timeout.create ~f` without introducing an anon fn, but I guess it's kind of silly to do it that way.

**Pull request merged**